src/lib.rs

```diff
@@ -313,6 +321,32 @@ fn restore_extended_key(
     })
 }
 
+fn derive_child_key_pair(
+    str_xprv: String,
+    str_path: String
+) -> Result<ChildKeyPair, Error> {
+    let secp = Secp256k1::new();
+    let xprv = ExtendedPrivKey::from_str(&str_xprv)?;
+    let path = DerivationPath::from_str(&str_path)?;
+    let derived_xprv = &xprv.derive_priv(&secp, &path)?;
+    let origin: KeySource = (xprv.fingerprint(&secp), path);
+    let derived_xprv_desc_key: DescriptorKey<Segwitv0> = 
+        derived_xprv.into_descriptor_key(Some(origin), DerivationPath::default())?;
+    if let Secret(desc_seckey, _, _) = derived_xprv_desc_key {
+        let desc_pubkey = desc_seckey
+            .as_public(&secp)
+            .map_err(|e| Error::Generic(e.to_string()))?;
```

Comment:
The `bdk-cli` code here uses `.unwrap()` - @notmandatory can you add some context?

Reply:
> The `bdk-cli` code here uses `.unwrap()` - @notmandatory can you add some context?

I think maybe you're looking at an older version of `bdk-cli` :-) this is the cli code works also without an `.unwrap()`.